docs/architecture/data-sources.md

```diff
@@ -0,0 +1,113 @@
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace CarbonAware.DataSources.MyNewDataSource.Configuration;
+
+public static class ServiceCollectionExtensions 
+{
+    public static void AddMyNewDataSource(this IServiceCollection services)
+    {
+        // ... register your data source with the IServiceCollection instance
+    }
+}
+```
+
+### Register the New Data Source
+Once the data source's ServiceCollectionExtensions is configured, it can be registered as an available data source for the SDK. This is done in the `CarbonAware.DataSources.Registration` project. In particular, there is an equivalent `ServiceCollectionExtensions` file in the `Configuration` directory of that project. The public function `AddDataSourceService` contains a switch statement that reads in the environment variables for the SDK and selects the requested data source. To register your new data source, you must: 
+1. Add a new enum type to the `DataSourceType` enum file in that Configuration folder.
+2. Add a new case statement in the `AddDataSourceService` function that matches on your new enum and adds your data source service to the services on the SDK (by calling your previously defined function).
+
+```csharp
+    switch (dataSourceType)
+    {
+        ...
+        case DataSourceType.MyNewDataSourceEnum:
+        {
+            services.AddMyNewDataSource();
+            break;
+        }
+        ...
+    }
+```
+
+### Adding Tests
+Each new data source is expected to come with a robust unit test suite that ensures that the main flows and edge cases are properly handled. This also ensures that the SDK can switch seamlessly between data sources and the experiences up the stack remains consistent and helpful to the user.
+
+The unit tests should be added as a new project under the data source's test directory: `CarbonAware.DataSources/CarbonAware.DataSources.MyNewDataSource/test`. Be sure to include a reference to the data source's project and add it to the solution. We have provided a command snippet below:
+
+```sh
+cd src
+dotnet new nunit --name CarbonAware.DataSources.MyNewDataSource.Tests -o CarbonAware.DataSources/CarbonAware.DataSources.MyNewDataSource/test
+dotnet sln CarbonAwareSDK.sln add CarbonAware.DataSources/CarbonAware.DataSources.MyNewDataSource/test/CarbonAware.DataSources.MyNewDataSource.Tests.csproj
+cd CarbonAware.DataSources/CarbonAware.DataSources.MyNewDataSource
+dotnet add tests/CarbonAware.DataSources.MyNewDataSource.Tests.csproj reference src/CarbonAware.DataSources.MyNewDataSource.csproj
+```
+### Try it Out!
+You are now ready to try out your new data source! Add a new environment variable that sets the data source and make sure to read it into the DataSource.Registration project. If you added a new CarbonIntensity data source, you can set the `CarbonIntensityDataSource` env variable:
```

Comment:
How does one `make sure to read it into the DataSource.Registration project` ?  I would recommend either adding directions or removing this line.  Also we should be clear about using the term `configuration setting` rather than `env var`.  `env vars` are one way to set `configuration settings`, but there are others.  The setting is what controls the app, not the `env var`.

```suggestion
You are now ready to try out your new data source!  If you added a new `ICarbonIntensityDataSource`, you can configure it using the `CarbonIntensityDataSource` setting:
```